File: port/assets/get_waves.js
const https = require('https');

const getWaves = function() {
	https.get("https://www.ndbc.noaa.gov/data/realtime2/46277.txt", function(res) {
	console.log(res.statusCode);
	res.setEncoding('utf-8');
	res.on('data', function(data) {
		const dataret = toString(data);
	console.log(dataret);
	
	});
	}).on('error', function(err){
		console.log(err);
	});
	return dataret;
};

const waveData = getWaves();
//console.log(waveData);

File: port/assets/get_waves.py
import requests

def set_buoy_url(buoy_url):
        buoy_url = buoy_url
        buoy = str(buoy_url)
        url = "https://www.ndbc.noaa.gov/data/realtime2/" + buoy + ".txt"
        return url 

def get_url_content(url):
        page = requests.get(url)
        content = page.content
        content = content.decode('utf-8')
        return content

if __name__ == "__main__":

        buoy_url = set_buoy_url(46277)
        page = get_url_content(buoy_url)
        print(page)